contributor_docs/documentation_style_guide.md

```diff
@@ -0,0 +1,1184 @@
+
+```javascript
+// Bad.
+//Remember this.
+const magicWord = 'Please';
+
+// Good.
+// Remember this.
+const magicWord = 'Please';
+```
+
+- Use `/** ... */` for multiline comments.
+
+```javascript
+// Bad.
+// I will not use // for multiline comments.
+// I will not use // for multiline comments.
+// I will not use // for multiline comments.
+// I will not use // for multiline comments.
+// I will not use // for multiline comments.
+
+// Good.
+/**
+ * I will not use // for multiline comments.
+ * I will not use // for multiline comments.
+ * I will not use // for multiline comments.
+ * I will not use // for multiline comments.
+ * I will not use // for multiline comments.
+ */
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Whitespace
+
+- Indent blocks 2 spaces.
+
+```javascript
+// Bad.
+function setup() {
+∙∙∙∙createCanvas(400, 400);
+}
+
+// Bad.
+function setup() {
+∙createCanvas(400, 400);
+}
+
+// Good.
+function setup() {
+∙∙createCanvas(400, 400);
+}
+```
+
+- Place 1 space before the leading brace.
+
+```javascript
+// Bad.
+function setup(){
+  createCanvas(400, 400);
+}
+
+// Good.
+function setup() {
+  createCanvas(400, 400);
+}
+```
+
+- Place 1 space before the opening parenthesis in control statements such as `if` and `for`. Place no space between the argument list and the function name.
+
+```javascript
+// Bad.
+if(keyIsPressed === true) {
+  doStuff ();
+}
+
+// Good.
+if (keyIsPressed === true) {
+  doStuff();
+}
+
+// Bad.
+function setup () {
+  createCanvas (400, 400);
+}
+
+// Good.
+function setup() {
+  createCanvas(400, 400);
+}
+```
+
+- Place spaces between operators.
+
+```javascript
+// Bad.
+const y=x+5;
+
+// Good.
+const y = x + 5;
+```
+
+## Semicolons
+
+- Yep.
+
+> Why? JavaScript's [automatic semicolon insertion](https://tc39.github.io/ecma262/#sec-automatic-semicolon-insertion) can lead to subtle bugs.
+
+```javascript
+// Bad.
+let x = 0
+
+// Good.
+let x = 0;
+```
+
+## Naming Conventions
+
+- Avoid single letter names. Be descriptive.
+
+```javascript
+// Bad.
+function f(x, y) {
+  // ...
+}
+
+// Good.
+function vectorField(x, y) {
+  // ...
+}
+```
+
+- Name objects, functions, and instances using camelCase.
+
+```javascript
+// Bad.
+const OBJEcttsssss = {};
+
+// Bad.
+const this_is_my_object = {};
+
+// Good.
+const thisIsMyObject = {};
+```
+
+- Name classes using PascalCase.
+
+```javascript
+// Bad.
+class player {
+  constructor(name) {
+    this.name = name;
+  }
+}
+
+// Good.
+class Player {
+  constructor(name) {
+    this.name = name;
+  }
+}
+```
+
+- Don't use trailing or leading underscores.
+
+> Why? JavaScript doesn't have private properties or methods.
+
+```javascript
+// Bad.
+class Spy {
+  constructor(secret) {
+    this._secret = secret;
+  }
+}
+
+// Good.
+class Spy {
+  constructor(secret) {
+    this.secret = secret;
+  }
+}
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Variables
+
+- Always use `const` or `let` to declare variables.
```

Comment:
Shall we include rules about the usage of 'var'?

Reply:
Good suggestion. "Explicit is better than implicit".